### PR TITLE
Add one case of virtual network

### PR DESCRIPTION
**Pull request by yanglei-rh**
1.Reboot guest multiple times with multiple NICs
ID: 1805551

Signed-off-by: Lei Yang <leiyang@redhat.com>

### CODE REVIEW

**Comment by yanglei-rh** [comment]
@PaulYuuu @kimi1978wy @xutian Please help me review it.Thanks a lot.

**Comment by PaulYuuu** [comment]
My suggestion is to refer to `reboot.cfg`, you can add a new cfg but use `boot.py`

**Comment by yanglei-rh** [comment]
@PaulYuuu If I use boot.py,I need to change the file content to satisfy multiple nics.I afraid this will affect other cases.

File: qemu/tests/cfg/multi_nics_reboot.cfg [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+            device_name = virtio-net-pci
+        - e1000:
+            RHEL:
+                only RHEL.6 RHEL.7
+            no RHEL.8
+            device_name = e1000
+        - e1000e:
+            device_name = e1000e
+        - rtl8139:
+            RHEL:
```
Comment (xutian) [inline qemu/tests/cfg/multi_nics_reboot.cfg]: no RHEL.8  is enough in here.

File: qemu/tests/multi_nics_reboot.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    5. Log into the guest to verify it's up again
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment.
+    """
+    # Get the ethernet cards number from params
+    nics_num = int(params.get("nics_num"))
+    for i in range(nics_num):
+        nics = "nic%s" % i
+        params["nics"] = ' '.join([params["nics"], nics])
```
Comment (xutian) [inline qemu/tests/multi_nics_reboot.py]: params["nics"] =  " ".join( list(["nic%s" % i for i in range(nics_num)]))

File: qemu/tests/multi_nics_reboot.py [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+    vm = env.get_vm(params["main_vm"])
+    vm.verify_alive()
+    logging.info("[ %s ] NICs card specified in config file" % nics_num)
+    session = vm.wait_for_login(timeout=int(params.get("login_timeout", 360)))
+    if params.get("reboot_method"):
+        # Reboot the VM
+        for i in range(int(params.get("reboot_count", 1))):
+            session = vm.reboot(session, params["reboot_method"], 0)
+            session.close()
```
Comment (xutian) [inline qemu/tests/multi_nics_reboot.py]: destroy vm too.

**Comment by PaulYuuu** [comment]
> @PaulYuuu If I use boot.py,I need to change the file content to satisfy multiple nics.I afraid this will affect other cases.

Not need. You just need to add `nics = nic1 nic2 nic3 nic4` in cfg

File: qemu/tests/multi_nics_reboot.py (hunk 3 above)
Comment (yanglei-rh) [reply qemu/tests/multi_nics_reboot.py]: I have modified the above three problems,thanks a lot.

**Comment by kimi1978wy** [comment]
I agree with PaulYuuu , it should in reboot.cfg is better.

**Comment by kimi1978wy** [comment]
It always boot with virtio-net pci with different variants.
-device virtio-net-pci,mac=9a:fc:93:d9:2a:3c,id=idYcClZB,netdev=idaVooiZ,bus=pcie-root-port-3,addr=0x0  \

**Comment by yanglei-rh** [comment]
If so,I will delete all the following variables,and specify the nic to be tested by "--nic_modle=xx" during the test.

**Comment by xutian** [comment]
Hi List
boot is very basic test, it call as sub test by many other tests, I think it keep simple and stable is  good idea.

**Comment by kimi1978wy** [comment]
(1/1) Host_RHEL.m8.u2.product_rhel.qcow2.virtio_blk.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.multi_nics_reboot.q35: PASS (1146.13 s)

File: qemu/tests/cfg/multi_nics_reboot.cfg [hunk 4]
```diff
@@ -0,0 +1,10 @@
+- multi_nics_reboot:
+    virt_test_type = qemu libvirt
```
Comment (PaulYuuu) [inline qemu/tests/cfg/multi_nics_reboot.cfg]: If the `virt_test_type` is `qemu libvirt`, I think this file should be moved to `generic`, not `qemu`
Other LGTM.

Comment (yanglei-rh) [reply qemu/tests/cfg/multi_nics_reboot.cfg]: I have deleted it.thanks

**Comment by PaulYuuu** [comment]
Acked-by: Yihuang Yu <yihyu@redhat.com>

**Pull request merged**